### DISCUSSION TITLE
Display API Response on page.

### DISCUSSION THREAD

**Discussion by tokkisva** [Budibase/budibase#9970] [Help] [answered]
Hi BB friends,

I have a button where 'On Click' it executes a POST API Query to another system- that system sends a response back with some extra data on what was submitted. I am looking for a way to display that raw response information on page for diagnostic purposes for my non-technical users. 

I saw an old post that referenced this KB article but it is dead:
https://docs.budibase.com/docs/referencing

Any thoughts on how to do this? I can find the response in dev console, just not seeing a way to show it on page in BB.

Thanks!

**Comment by andz-bb** [comment, marked as answer]
hey @tokkisva 

This might be the relevant page in the docs: https://docs.budibase.com/docs/chaining-actions#action-referencing

You can certainly show the API response in your screen. All you need to do is run the query action and then set the result for that action in your [App State](https://docs.budibase.com/docs/app-state).

<img width="941" alt="setting response to state" src="https://user-images.githubusercontent.com/110921612/224162988-87b568bf-7859-47ed-acf2-035dddec40e0.png">

Once it is added to your state you can use the state variable as a binding in another component like so:

<img width="939" alt="state binding" src="https://user-images.githubusercontent.com/110921612/224163157-d761615b-6e97-428d-b685-00cdd788659a.png">

Now inside my app the full POST response is displayed whenever the button is clicked.

<img width="1236" alt="response displayed" src="https://user-images.githubusercontent.com/110921612/224162897-d662c135-710d-443e-8435-41fb730a4e62.png">

Hope this helps!

**Reply by tokkisva** [reply]
This is exactly what I was looking for, thanks!!
